### PR TITLE
Update to solve Portugese language bug.

### PR DESCRIPTION
**Pull request by Hazlitte**
## Summary

Update to solve Portugese language bug.

Addresses issue #2105 

## Checklist

- [x] My code is tested and passes existing unit tests.
- [ ] My code has an appropriate set of unit tests which all pass.
- [ ] My code is backward-compatible with WordPress 4.7 and PHP 5.6.
- [ ] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [ ] My code has proper inline documentation.
- [X] I have added a QA Brief on the issue linked above.
- [X] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

### CODE REVIEW

File: includes/Context.php [hunk 1]
```diff
@@ -467,4 +467,36 @@ private function filter_reference_url( $url = '' ) {
...
+		// Return locale in the required format.
+		if ( 'language-code' === $format ) {
+			$code_array = explode( '_', $wp_locale );
+			return $code_array[0];
+
+		} elseif ( 'language-variant' === $format ) {
+			$variant_array  = explode( '_', $wp_locale );
+			$variant_string = array_key_exists( 1, $variant_array ) ? $variant_array[0] . '_' . $variant_array[1] : $variant_array[0];
+			return $variant_string;
+		}
```
Comment (eugene-manuilov) [inline includes/Context.php:498]: @Hazlitte let's add phpunit tests to make sure this logic works correctly. You can change the global locale using something like this:

```php
$original_locale = $GLOBALS['locale'];
$GLOBALS['locale'] = 'pt_PT_ao90';

...

$GLOBALS['locale'] = $original_locale;
```

File: includes/Context.php [hunk 2]
```diff
@@ -467,4 +467,36 @@ private function filter_reference_url( $url = '' ) {
...
+
+		// Return locale in the required format.
+		if ( 'language-code' === $format ) {
+			$code_array = explode( '_', $wp_locale );
+			return $code_array[0];
+
+		} elseif ( 'language-variant' === $format ) {
+			$variant_array  = explode( '_', $wp_locale );
+			$variant_string = array_key_exists( 1, $variant_array ) ? $variant_array[0] . '_' . $variant_array[1] : $variant_array[0];
```
Comment (eugene-manuilov) [inline includes/Context.php:496]: I think we can simplify it like this:

```suggestion
			$variant_string = implode( '_', array_slice( $variant_array, 0, 2 ) );
```

**Review by eugene-manuilov — CHANGES_REQUESTED** [review]
Nice work, @Hazlitte! Let's add phpunit tests for the new method and it should be good to go.

File: includes/Context.php (hunk 1 above)
Comment (Hazlitte) [reply includes/Context.php:498]: 👍

**Review by eugene-manuilov — APPROVED** [review]
Awesome! Thanks, @Hazlitte!

**Pull request merged**